cellmlmanip/_singularity_fixes.py

```diff
@@ -0,0 +1,234 @@
+"""
+`remove_fixable_singularities` specifies a method to remove fixable singularities from a model
+This represents a flattened CellML model and metadata about its variables.
+"""
+from math import isclose
+
+import networkx as nx
+from sympy import (
+    Abs,
+    Add,
+    Eq,
+    Float,
+    I,
+    Mul,
+    Piecewise,
+    Pow,
+    Wild,
+    exp,
+    log,
+    solveset,
+)
+
+import cellmlmanip
+from cellmlmanip.model import FLOAT_PRECISION
+
+
+ONE = cellmlmanip.Quantity(1.0, 'dimensionless')
+
+
+def _generate_piecewise(vs, ve, sp, ex, V):
+    """Generates a piecewise for expression based on the singularity point (sp) and vmin (vs) and vmax (ve) """
+    if vs is None:  # This shouldn't be apiecewise since we have no Vmin / Vmax
+        return ex
+
+    f_vs = ex.xreplace({V: vs})
+    f_ve = ex.xreplace({V: ve})
+    return Piecewise((f_vs + ((V - vs) / (ve - vs)) * (f_ve - f_vs),
+                      Abs(V - sp) < Abs((ve - vs) / 2)), (ex, True))
```

Comment:
Why when `|V - V_singularity|` is less than half the range from ve to vs ? I thought you'd use ve and vs as the cut-off points so that it's nice and continuous?

Reply:
vs = solve (U-1e-7) for V
ve = vs = solve (U+1e-7) for V
sp = solve (U) for V
so ve - vs is the whole range 1e-7 either side of sp or if we substract sp:
1e-7 - -1e-7 = 2e-7 so that's twice what we want hence the half

Reply:
so why not just write that into the equation instead of using 3 variables?

Reply:
Oh wait, I don't understand what `solve` means in your example

Reply:
with solve (U+1e-7) for V I mean:
`solveset(u + 1e-7, V)`
that code gets me the value of V for which (U + 1e-7) == 0

Reply:
> so why not just write that into the equation instead of using 3 variables?

how would I write that?